src/util/adb.js

```diff
@@ -271,8 +271,15 @@ export default class ADBUtils {
       value: `-profile ${deviceProfileDir}`,
     }];
 
-    const component = apkComponent ?
-      `${apk}/.${apkComponent}` : `${apk}/.App`;
+    let component;
+
+    if (!apkComponent) {
+      component = `${apk}/.App`;
+    } else if (apkComponent.includes('.')) {
+      component = `${apk}/${apkComponent}`;
+    } else {
+      component = `${apk}/.${apkComponent}`;
```

Comment:
Prepend a comment:

```
// adb expands the following to: `${apk}/${apk}.${apkComponent}`
```

Reply:
Just to be sure, would you like this comment after the line 278  `else if` statement?

Reply:
It kind of applies to every line indeed. Let's modify the apkComponent variable if needed and add the comment before `component`, like this:

```javascript
if (!apkComponent) {
  apkComponent = ".App";
} else if (!apkComponent.includes(".")) {
  apkComponent = '.' + apkComponent;
}
// if `apkComponent` starts with a '.', then adb will expand
// the following to: `${apk}/${apk}.${apkComponent}`
const component = `${apk}/${apkComponent}`;
```

Reply:
That looks really good - define the `component` after the `apkComponent` check, I like it